### DISCUSSION TITLE
tp.file.exists() == false always returns false

### DISCUSSION THREAD

**Discussion by ilandikov** [SilentVoid13/Templater#847] [Help] [answered]
Hello everyone

I'm seeking help with my periodic notes. I with to create weekly and monthly notes automatically from my daily notes. So I use the following code in the daily template:
<%*
    if (tp.file.exists(tp.date.now("YYYY[-W]WW", 0)) == false) {
        (await tp.file.create_new(tp.file.find_tfile("\_weekly"), tp.date.now("YYYY[-W]WW")))
-%>

- [ ] Set weekly targets [[<% tp.date.now("YYYY[-W]WW", 0) %>]]
<%*
    }
-%>
<%*
    if (tp.file.exists(tp.date.now("YYYY-MM", 0)) == false) {
        (await tp.file.create_new(tp.file.find_tfile("\_monthly"), tp.date.now("YYYY-MM")))
-%>

- [ ] Set monthly targets [[<% tp.date.now("YYYY-MM", 0) %>]]
<%*
    }
-%>

The weekly creation works perfectly. The monthly doesn't =( I tried to console.log(tp.file.exists(tp.date.now("YYYY-MM", 0)) == false) and is it always returning false.

Am I doing something wrong?
Is there another and/or more elegant way to do this?

Obisidian 0.15.9
Templater 1.14.3

Thanks
Ilyas

**Comment by AB1908** [comment, marked as answer]
Have you tried logging `tp.file.exists()`? I think it switched to `async` recently and you probably need to `await` the `Promise`.

**Comment by ilandikov** [comment]
Oh...I haven't notice that, guess the documentation has to be updated as well.

Thanks for the tip, I checked the return and it was a Promise indeed. Now I switched to await but the result is always false still.Could you help me construct the request?

For instance, I have a file "2022-W39.md". I tried:
console.log(await tp.file.exists("\[\["+
tp.date.now("YYYY[-W]WW", 0)+"\]\]"))
console.log(await tp.file.exists(tp.date.now("YYYY[-W]WW", 0)))
console.log(await tp.file.exists("\""+tp.date.now("YYYY[-W]WW", 0)+"\""))
console.log(await tp.file.exists("\""+tp.date.now("YYYY[-W]WW", 0)+".md\""))

But I get false every time

**Reply by AB1908** [reply]
Can you log the output of `tp.date.now()`? Perhaps the output is not in the format you expect.

**Reply by ilandikov** [reply]
I tried that before and the format is correct linke . In which format does the tp.file.exists() shall take input?

I event tried tp.file.exists("existing_file_name"), tp.file.exists("existing_file_name.md") and tp.file.exists([[existing_file_name]]) and they all return false =/

**Reply by ilandikov** [reply]
Ok I got the answer now =) I also need to add the path and the extension and then it works eg "folder/folder/filename.md"

Shouldn't there be a shorter way?

**Reply by AB1908** [reply]
It's probably a wrapper around an Obsidian API that behaves in a similar way.

**Reply by shabegom** [reply]
Yeah it just wraps a built in `exists()` method. The full path is needed (and intentional) because there could be multiple files with the same basename across different parts of a vault.

**Reply by shabegom** [reply]
The one thing we could do is make it relative to the path of where the command is fired. If that's useful you can file an enhancement request in the Issues

**Reply by ilandikov** [reply]
I agree on the problem of multiple files with the same basename, but shouldn't there be a shorter way to do this...

For now I'm using (await tp.file.exists(tp.file.folder(false) + "/" + tp.date.now("YYYY-MM", 0) + ".md")) and for me this doesn't look pretty. I will file a request! Thanks a lot! =)
